File: AssetManagement.Contracts/Dtos/AssetDtos/Responses/AssetDetailsResponse.cs
namespace AssetManagement.Contracts.Dtos.AssetDtos.Responses
{
    public class AssetDetailsResponse
    {
        public string? Id { get; set; }
        public string? AssetCode { get; set; }
        public string? AssetName { get; set; }
        public string? CategoryName { get; set; }
        public DateTime InstalledDate { get; set; }
        public string? State { get; set; }
        public string? Location { get; set; }
        public string? Specification { get; set; }
    }
}

File: AssetManagement.Application/Frontend/src/app/models/asset/AssetDetailsResponse.ts
interface AssignmentResponse {
  date: Date;
  assignedTo: string;
  assignedBy: string;
  returnedDate: Date;
}
interface AssetDetailsResponse {
  assetId: string;
  assetCode: string;
  assetName: string;
  categoryName: string;
  installedDate: Date;
  state: string;
  location: string;
  specification: string;
  assignments: Array<AssignmentResponse>;
}

export default AssetDetailsResponse;
